### PR TITLE
Hero Component (WIP)

### PR DESCRIPTION
**Pull request by gvn**

### CODE REVIEW

File: content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx [hunk 1]
```diff
@@ -17,7 +17,10 @@ export class _DiscoveryStreamBase extends React.PureComponent {
       case "CardGrid":
         return (<CardGrid />);
       case "Hero":
-        return (<Hero />);
+        return (<Hero
+          feed={component.feed.url}
+          style={component.properties.style}
```
Comment (dmose) [inline content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx]: Is this intended to be on every component as the way to override the local style from the server?  Or is it something else?

Comment (gvn) [reply content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx]: I'm basing this off [this specification](https://docs.google.com/document/d/1RCVxPGY2EbOr3tG19MIpfAaA59ZOApt303XUI0-RMoM/edit#), which is what the [endpoint schema](https://app.swaggerhub.com/apis/PocketNewTab/FirefoxHome/0.2) is derived from. 

In this case it's just controlling whether or not the component has a border.

Comment (gvn) [reply content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx]: PS: There's a lot more coming in this patch. Not ready for review yet, but questions are totally fine! :)

File: content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx [hunk 2]
```diff
@@ -2,14 +2,37 @@ import {connect} from "react-redux";
 import React from "react";
 
 export class _Hero extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    const feed = this.props.DiscoveryStream.feeds[this.props.feed.url];
+
+    this.state = {
+      recommendations: feed.data.recommendations
```
Comment (ScottDowne) [inline content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx]: I dunno what happens if you put something stored in redux into a component's state. Example, if the redux store changes, I don't know what happens to the data in the component's state. Might work, dunno.

Comment (gvn) [reply content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx]: What's the alternative?

Comment (Mardak) [reply content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx]: Typically just read the data from `props`

**Comment by gvn** [comment]
@ScottDowne you can go ahead and review so we can get a rough version landed and do a follow-on ticket for the remaining work.

**Comment by gvn** [comment]
@ScottDowne [This is the faked endpoint](https://gist.githubusercontent.com/gvn/76af39c92085c247dd5dd7ccc2412618/raw/0366a32853b35244807d9b90b1becb61157ccef9/disco.json) I used for testing the 3 layout widths.

**Comment by gvn** [comment]
Stuff remaining to do:

- fix image aspect ratios
- wire up links
- generally tighten up styling
- set count of cards based on remote layout definition

File: content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx [hunk 3]
```diff
@@ -1,15 +1,51 @@
 import {connect} from "react-redux";
+import {DSCard} from "../DSCard/DSCard.jsx";
```
Comment (ScottDowne) [inline content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx]: It looks like most components only use relative links if it's `./` and it also looks like the .jsx is always omitted.

I don't personally have an opinion, just noticing this.

**Pull request merged**